apps/antalmanac/src/actions/ActionTypesStore.ts

```diff
@@ -0,0 +1,151 @@
+import { EventEmitter } from 'events';
+import { RepeatingCustomEvent, ScheduleCourse } from '@packages/antalmanac-types';
+import { saveSchedule } from '$actions/AppStoreActions';
+import AppStore from '$stores/AppStore';
+
+export interface BaseAction {
+    type: string;
+}
```

Comment:
```suggestion
```

There's no reason for the base action. The type discrimination will work without it, and the reason this type of interface would exist is to ensure classes adhere to a basic set of functionality via the `implements` (TypeScript) directive. For example

```tsx
interface BaseClass {
  doSomething: () => any
}

class MyExtendedClass implements BaseClass {
  // TS will display an error if you don't have a `doSomething` method. 
}
```